src/extension.ts

```diff
@@ -1,138 +1,65 @@
-import {
-    activateExtension,
-    DefinitionRequest,
-    DidOpenTextDocumentNotification,
-    DidOpenTextDocumentParams,
-    HoverRequest,
-    ImplementationRequest,
-    ReferencesRequest,
-    Registration,
-    RegistrationParams,
-    RegistrationRequest,
-    SourcegraphExtensionAPI,
-    TextDocumentPositionParams,
-    TypeDefinitionRequest,
-} from 'sourcegraph'
-import { createWebWorkerMessageTransports } from 'sourcegraph/module/jsonrpc2/transports/webWorker'
-import { Hover } from 'vscode-languageserver-types'
+import * as sourcegraph from 'sourcegraph'
+import { toHover } from './backcompat'
 import { sendLSPRequest } from './lsp'
-import { getModeFromPath } from './temporaryModes'
 
 interface Settings {
     ['languageServer.url']: string
 }
 
-let loggedNoURL = false
-
-/**
- * Fixes a response to textDocument/hover that is invalid because either
- * `range` or `contents` are `null`.
- *
- * See the spec:
- *
- * https://microsoft.github.io/language-server-protocol/specification#textDocument_hover
- *
- * @param response The LSP response to fix (will be mutated)
- */
-const normalizeHoverResponse = (hoverResult: any): void => {
-    // rls for Rust sometimes responds with `range: null`.
-    // https://github.com/sourcegraph/sourcegraph/issues/11880
-    if (hoverResult && !hoverResult.range) {
-        hoverResult.range = undefined
-    }
-
-    // clangd for C/C++ sometimes responds with `contents: null`.
-    // https://github.com/sourcegraph/sourcegraph/issues/11880#issuecomment-396650342
-    if (hoverResult && !hoverResult.contents) {
-        hoverResult.contents = []
-    }
+/** Entrypoint for the language server HTTP adapter Sourcegraph extension. */
+export function activate(): void {
+    sourcegraph.languages.registerHoverProvider(['*'], {
+        provideHover: async (doc, pos) => {
```

Comment:
I would find this code more readable if this was `textDocument, position`. I know, it's 15 characters more. But personally my reading flow always gets interrupted when I meet these abbreviations.